src/main.c

```diff
@@ -36,16 +39,29 @@ record_env(FILE *fout, char **ep)
 	(void) fprintf(fout, "%s\n", *p);
 }
 
+char **
+handle_options(char **argv)
```

Comment:
No need for a separate function, add the code in `main()`.
This way you won't be returning changes, but only modifying the local variable `argv`